GoogleForCreators/web-stories-wp
# Dashboard: Implement updated snackbar from design system

## Summary
Formerly called "toasts" in dashboard and implemented to fill a need for error messaging, now that we have updated designs for a [snackbar from Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3398%3A83619), this PR swaps out the old structure in the dashboard for those new ones. 

## Relevant Technical Choices

Clean up: 
- Since we were calling these toasts and stacking them in the dashboard and now we are formalizing our designs and calling them snackbars and limiting to displaying one message at a time, I did rename everything in the dashboard from toast to snackbar. That's the bulk of these updates. 
- Moved snackbar context for dashboard up to `dashboard/app/snackbar` 
- `<DashboardSnackbar />`, the view of the snackbar in the dashboard, is now imported and shown in the `<Layout.Fixed />` portion of views. This preserves scrolling from the lower section of the view when a snackbar message is present and also prevents interference of the message with, say, a context menu button getting clicked.  

Functionality:
- Had to make a few updates to the design-system snackbar that I didn't see when testing just in storybook for that - namely, the `z-index` and `fixed` position should be on the message, particularly now that there's only 1 being displayed at a time. 
- Updated `snackbarProvider.js` to queue the messages one at a time and clear message queue when `currentPath` from router updates.

## To-do

- Since the designs for the snackbar only have a message, we should take a second look at the error messages used in the dashboard, they were all relying on having a title present. I have this pr open: #5305 to organize the static text for errors. I think a follow up to this would be beneficial. 

## User-facing changes

(In dashboard)
- Snackbar messages should be updated to look like the new ones in figma: https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3398%3A83619
- Only one message should ever show at once 
- Messages are not able to be early dismissed 

## Testing Instructions

- Things that have snackbar messages on the dashboard: 
  - Copy a story URL from a context menu 
  - Any API error (I just block the domain after initial load and click around) 
- You can also see that only 1 message occurs at once by playing with the demo that's tied to context in storybook (dashboard/views/dashboardSnackbar)


<img width="1359" alt="Screen Shot 2020-11-23 at 9 06 07 PM" src="https://user-images.githubusercontent.com/10720454/100047454-c0bca200-2dcf-11eb-86d2-f4d41b8689c3.png">


---



Addresses #4967
